library/std/src/thread/mod.rs

```diff
@@ -1203,6 +1203,50 @@ impl Thread {
         self.cname().map(|s| unsafe { str::from_utf8_unchecked(s.to_bytes()) })
     }
 
+    /// Consumes the `Thread`, returning a raw pointer.
```

Comment:
You might want to make the guarantees of this pointer more explicit, for example:

* It can’t be round-tripped though `&()`
* Any kind of reading or writing to the address of this pointer should be considered unsound
* The pointer is not valid for anything other than `Thread::from_raw`

Reply:
> It can’t be round-tripped though `&()`

I don't think this should be documented here. Maybe it should go on the `pointer` type itself, because you should never expect to be able to round-trip a pointer to any type through a reference of any type.

Reply:
Well, it would be fine to expect it for `<NonNull<()>>::dangling()`. But it’s still a fair point.

Reply:
Making the reference for that case is UB